src/main/java/io/managed/services/test/client/kafkainstance/KafkaInstanceApi.java

```diff
@@ -1,115 +1,148 @@
     }
 
     @Override
     protected ApiUnknownException toApiException(Exception e) {
-        if (e instanceof ApiException) {
-            var ex = (ApiException) e;
-            return new ApiUnknownException(ex.getMessage(), ex.getCode(), ex.getResponseHeaders(), ex.getResponseBody(), ex);
+        log.info(e);
+
+        if (e.getCause() != null) {
+            if (e.getCause() instanceof com.openshift.cloud.api.kas.auth.models.Error) {
+                var err = (com.openshift.cloud.api.kas.auth.models.Error) e.getCause();
+                return new ApiUnknownException(err.getMessage(), err.getCode(), new HashMap<>(), "", err);
+            }
+            if (e.getCause() instanceof com.openshift.cloud.api.kas.models.Error) {
+                var err = (com.openshift.cloud.api.kas.models.Error) e.getCause();
+                return new ApiUnknownException(err.getMessage(), e.hashCode(), new HashMap<>(), "", err);
+            }
+
         }
-        return null;
-    }
 
-    @Override
-    protected void setAccessToken(String accessToken) {
-        this.apiClient.setAccessToken(accessToken);
+        return null;
     }
 
 
     public Topic updateTopic(String name, TopicSettings ts) throws ApiGenericException {
         //return getTopics(null, null, null, null, null);
-        return retry(() -> topicsApi.updateTopic(name, ts));
+        return retry(() -> v1.topics(name).patch(ts).get(10, TimeUnit.SECONDS));
     }
 
     public TopicsList getTopics() throws ApiGenericException {
         return getTopics(null, null, null, null, null);
     }
 
-    public TopicsList getTopics(Integer size, Integer page, String filter, SortDirection order, TopicOrderKey orderKey) throws ApiGenericException {
-        return retry(() -> topicsApi.getTopics(size, filter, page, order, orderKey));
+    public TopicsList getTopics(Integer size, Integer page, String filter, String order, String orderKey) throws ApiGenericException {
+        return retry(() -> v1.topics().get(config -> {
+            config.queryParameters.size = size;
+            config.queryParameters.page = page;
+            config.queryParameters.filter = filter;
+            config.queryParameters.order = order;
+            config.queryParameters.orderKey = orderKey;
+        }).get(10, TimeUnit.SECONDS));
     }
 
     public Topic getTopic(String topicName) throws ApiGenericException {
-        return retry(() -> topicsApi.getTopic(topicName));
+        return retry(() -> v1.topics(topicName).get().get(10, TimeUnit.SECONDS));
     }
 
     public Topic createTopic(NewTopicInput newTopicInput) throws ApiGenericException {
-        return retry(() -> topicsApi.createTopic(newTopicInput));
+        return retry(() -> v1.topics().post(newTopicInput).get(10, TimeUnit.SECONDS));
     }
 
     public void deleteTopic(String topicName) throws ApiGenericException {
-        retry(() -> topicsApi.deleteTopic(topicName));
+        retry(() -> v1.topics(topicName).delete());
```

Comment:
Doesn't seem to be returning error when deleting a non-existent topic.

Reply:
isn't it expected?

Reply:
Tests expect it to throw error, I think 404 should be thrown, let me check again.

Reply:
Delete topic should return 404 for non-existent topic.

Reply:
What's the actual response?

Reply:
```
HTTP/1.1 404 Not Found
Content-Length: 119
Content-Type: application/json;charset=UTF-8
Strict-Transport-Security: max-age=31536000

{"id":"3","kind":"Error","href":"/api/v1/errors/3","reason":"No such topic","code":404,"error_message":"No such topic"}
2023/03/15 14:36:56 
HTTP/1.1 404 Not Found
Content-Length: 119
Content-Type: application/json;charset=UTF-8
Strict-Transport-Security: max-age=31536000
```

Reply:
Everything looks ok from the answer, am I missing something?

Reply:
@rkpattnaik780 tested, and I cannot reproduce, when deleting a non-existing topic an `ExecutionException` is thrown with `Cause` a `com.openshift.cloud.api.kas.auth.models.Error` object, with code `404` (as expected) and `ErrorMessage` and `Reason` set to: `No such topic`.

Do you have enough elements to provide a fix for this?

Reply:
change the line please to

```
retry(() -> v1.topics(topicName).delete().get(10, TimeUnit.SECONDS));
```